src/renderer/dx/DxRenderer.cpp

```diff
@@ -925,6 +943,21 @@ void DxEngine::_InvalidOr(RECT rc) noexcept
     FAIL_FAST_IF_FAILED(InvalidateAll());
     RETURN_HR_IF(E_NOT_VALID_STATE, _isPainting); // invalid to start a paint while painting.
 
+    TraceLoggingWrite(g_hDxRenderProvider,

```

Comment:
Should we have the same perf thing from #4594 here?

Reply:
Ah, sure, OK.

Reply:
TraceLoggingWrite is inert if it's not enabled -- it's only computational logic _outside_ of a TraceLoggingWrite that needs to be guarded

Reply:
I mean it skips all the division that I'm doing, which arguably isn't much... but it's something.

Reply:
OK no, I think the macro hides it all because it was inside the macro. Rolling back.